src/sentry/sentry_apps/utils/alert_rule_action.py

```diff
@@ -0,0 +1,117 @@
+from collections.abc import Mapping, Sequence
+from typing import Any
+
+import sentry_sdk
+from rest_framework import serializers, status
+from rest_framework.response import Response
+
+from sentry.auth.access import NoAccess
+from sentry.constants import SENTRY_APP_ACTIONS
+from sentry.incidents.logic import get_filtered_actions
+from sentry.incidents.models.alert_rule import AlertRuleTriggerAction
+from sentry.incidents.serializers import AlertRuleTriggerActionSerializer
+from sentry.sentry_apps.services.app import app_service
+from sentry.sentry_apps.services.app.model import RpcAlertRuleActionResult
+from sentry.sentry_apps.utils.errors import (
+    SentryAppError,
+    SentryAppErrorType,
+    SentryAppIntegratorError,
+)
+
+
+def raise_alert_rule_action_result_errors(result: RpcAlertRuleActionResult) -> None:
+    if result.error_type is None:
+        return None
+
+    error_type = SentryAppErrorType(result.error_type)
+    match error_type:
+        case SentryAppErrorType.INTEGRATOR:
+            raise SentryAppIntegratorError(result.message)
+        case SentryAppErrorType.CLIENT:
+            raise SentryAppError(result.message)
+        case SentryAppErrorType.SENTRY:
+            raise Exception(result.message)
+
+
+def create_sentry_app_alert_rule_component_for_incidents(
```

Comment:
It is confusing to me that one of the create component methods you have here returns a component and one of them return an error and None on success.

Reply:
[1] I don't see much benefit in this method. How about you directly call `trigger_sentry_app_action_creators_for_incidents` and let the exception go through.